### ISSUE TITLE
Me Capability Adaptor

### ISSUE THREAD

**Issue by IsaacWorkman** [Darkosto/SkyFactory4Issues#851]
## Context
Right now, the only way to access AE2 across Compact Machines is to use quantum bridges. In the compact sky map mode though you can quickly rack up a dozen different compact machines, so quantum bridges are simply too expensive to maintain. 

I think the ME Capability Adapter mod would solve this pretty well. 

## Mod Details
* Name: ME Capability Adaptor
* CurseForge Link: https://minecraft.curseforge.com/projects/capability-adapter

**Comment by LezChap** [comment]
I agree if this mod doesn't introduce bugs, it would be a good addition.  Right now the only solutions to connect ME networks inside and outside a Compact Machine are to create subnets through the Compact Machine Walls/Tunnel interface, or to use the Quantum Bridges.  Neither are great solutions and have undesirable limitations (Quantum Bridges take up a 3x3x1 space in an already limited space and have a decent energy cost, subnets restrict crafting and storage abilities between the networks).

**Comment by Durandle** [comment]
can't you connect them via regular cable and the compact machines tunnels? i know most pipes/ducts work fine via the tunnels and refined storage the "better" version of the ME system works fine with them last i tried

**Comment by LezChap** [comment]
My understanding of the coding/technical aspect of your response, Durandle:
Tunnels only allow connections that implement Forge Capabilities.  Most mod's pipes do, as it lets Forge do the heavy lifting for simple things like moving fluids, items, and energy.  Some other mod pipes, like Logic Cables from Integrated Dynamics, and Refined Storage networks implement their network connections/cables as a Capability too, and thus connect.  AE2 doesn't implement them as Force Capabilities.  The only thing in AE2 (afaik) that implements them, is the Subnets.

And Refined Storage has it's issues, just like AE2.  I wouldn't call it "better"...in some respects, it's far worse.  It's just less complex for a player to figure out, and most people tend call that "better".  Different strokes for different folks and all that.

**Comment by IsaacWorkman** [comment]
You can "wrap" a compact machine as a sub network by using storage busses and interfaces, but it doesn't allow them to be on the same network. This is fine for some tasks (such as storage) but doesn't work for things such as crafting requests. Which is bad if you're trying to isolate network functionality.

**Comment by tolan1** [comment]
Yeah, the compact sky map type would be improved greatly by the addition of capability adapters.  Even in other maps, it very nice to have parts of the network in a compact machine.

**Comment by desagas** [comment]
I would love to see this mod added. Capability adapters would make automation, especially in a compactsky so much more organized.

**Comment by rosshadden** [comment]
I added the Capability Adapter to this pack based on your recommendation and it has been a great experience. Thank you for the recommendation, and :+1: for adding it to the pack.

**Comment by KollinsPlays** [comment]
As others have mentioned, other mods work with Compact Machines (CM) because they use Forge Capabilities and AE doesn't,. For mods like SSN, ID/IT, XNet, etc you just run a standard cable to 1 face of the CM and inside put a tunnel and another standard cable and although mods like SSN don't show the cable as connected, they will "see" each other as long as you are chunkloaded outside (CMs are chunkloaded inside automatically.) The Capability Adapter mod adds a Forge Capability block/adapter and you just put 1 on an outside face and 1 inside on a CM tunnel and run AE cable to each.

All the required mods like Forgelin are already in SF4 and I was able to add the capability adapter mod with no missing mods or conflicts. Now to figure out what the "disabled item" is in the recipe and if it's been removed from SF4 or just gated behind prestige...

**Comment by TheFinalDawn** [comment]
the 2 variants of Disabled Item are Dense and Smart cables.
